designer/fitting/dwipi.py

```diff
@@ -1285,32 +1294,53 @@ def outlierHelper(dwi, bmat, sigma, b, b0_pos, maxiter=25, convcrit=1e-3, levera
 
             # WLLS estimation
             # dt_i = np.linalg.lstsq(bmat_i, np.log(dwi_i), rcond=None)[0]
-            dt_i = np.linalg.solve(np.dot(bmat_i.T, bmat_i), np.dot(bmat_i.T, np.log(dwi_i)))
+            try:
+                dt_i = np.linalg.solve(np.dot(bmat_i.T, bmat_i), np.dot(bmat_i.T, np.log(dwi_i)))
+            except:
+                dt_i = minZero
             w = np.exp(np.matmul(bmat_i, dt_i))
             # dt_i = np.linalg.lstsq((bmat_i * np.tile(w, (1, nparam))), (np.log(dwi_i).reshape((dwi_i.shape[0], 1)) * w),
             #                        rcond=None)[0]
-            dt_i = np.linalg.solve(np.dot((bmat_i * np.tile(w, (1, nparam))).T, (bmat_i * np.tile(w, (1, nparam)))), \
-                                   np.dot((bmat_i * np.tile(w, (1, nparam))).T, (np.log(dwi_i).reshape((dwi_i.shape[0], 1)) * w)))
+            try:
+                dt_i = np.linalg.solve(np.dot((bmat_i * np.tile(w, (1, nparam))).T, (bmat_i * np.tile(w, (1, nparam)))), \
+                                    np.dot((bmat_i * np.tile(w, (1, nparam))).T, (np.log(dwi_i).reshape((dwi_i.shape[0], 1)) * w)))
+            except:
+                dwi_hat = minZero
             dwi_hat = np.exp(np.matmul(bmat_i, dt_i))
 
             # Goodness-of-fit
             residu = np.log(dwi_i.reshape((dwi_i.shape[0],1))) - np.log(dwi_hat)
             residu_ = dwi_i.reshape((dwi_i.shape[0],1)) - dwi_hat
-            chi2 = np.sum((residu_ * residu_) / np.square(sigma)) / (ndof_i) -1
-            gof = np.abs(chi2) < 3 * np.sqrt(2/ndof_i)
+            try:
+                chi2 = np.sum((residu_ * residu_) / np.square(sigma)) / (ndof_i) -1
+            except:
+                chi2 = minZero
+            try:
+                gof = np.abs(chi2) < 3 * np.sqrt(2/ndof_i)
+            except:
+                gof = True  # If ndof_i = 0, right inequality becomes inf and makes the logic True
             gof2 = gof
 
             # Iterative reweighning procedure
             iter = 0
             while ~gof and iter < maxiter:
```

Comment:
Is there some reason your using the bitwise NOT?

Reply:
Resolved in 5a0edaafb66338c83d93ef83c2989eb4220845c5